plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/DiagramPart.java

```diff
@@ -203,14 +205,16 @@ public void rearrangeDiagram()
         getFigure().repaint();
     }
 
-    private void resetConnectionConstraints(List sourceConnections) {
+    private void resetConnectionConstraints(List<?> sourceConnections) {
         if (!CommonUtils.isEmpty(sourceConnections)) {
```

Comment:
We can use real monitor. E.g. by creating a job in rearrangeDiagram and passing job's monitor.
Or by using runInProgreaaService

Reply:
done